### ISSUE TITLE
OMR_VMThread  Flags for GC barrier Awareness

### ISSUE THREAD

**Issue by RSalman** [eclipse/omr#6258]
Downstream code (OpenJ9) has thread flags to indicate Concurrent State / barrier awareness, e.g, 
`J9_PRIVATE_FLAGS_CONCURRENT_MARK_ACTIVE`. These should be implemented for OMR_VMThread.

There have been some attempts for generational barrier, but the work does not seem to be completed:

```C++
typedef struct OMR_VMThread {
...
	/* todo: dagar these are temporarily duplicated and should be removed from J9VMThread */
	void *lowTenureAddress;
	void *highTenureAddress;

	void *heapBaseForBarrierRange0;
	uintptr_t heapSizeForBarrierRange0; 
```

**Comment by RSalman** [comment]
cc @dmitripivkine cc @amicic

**Comment by RSalman** [comment]
I believe this can be tagged `good first issue` (@0xdaryl )

**Comment by shivam969619** [comment]
hey i am new here can i take this issue.

**Comment by RSalman** [comment]
Yup, go for it! You can reach out on the OMR slack channel if you'd like to chat about it, or feel free to post any questions here

**Comment by rawmeat21** [comment]
Is this issue still open? I'd like to work on this.

**Comment by 0xdaryl** [comment]
Perhaps @amicic can confirm the status of this feature.
